x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/metadata/Hyperparameters.java

```diff
@@ -0,0 +1,118 @@
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class Hyperparameters implements ToXContentObject, Writeable {
+
+    private static final String NAME = "hyperparameters";
+    public static final ParseField HYPERPARAMETER_NAME = new ParseField("name");
+    public static final ParseField VALUE = new ParseField("value");
+    public static final ParseField ABSOLUTE_IMPORTANCE = new ParseField("absolute_importance");
+    public static final ParseField RELATIVE_IMPORTANCE = new ParseField("relative_importance");
+    public static final ParseField SUPPLIED = new ParseField("supplied");
+
+
+    // These parsers follow the pattern that metadata is parsed leniently (to allow for enhancements), whilst config is parsed strictly
+    public static final ConstructingObjectParser<Hyperparameters, Void> LENIENT_PARSER = createParser(true);
+    public static final ConstructingObjectParser<Hyperparameters, Void> STRICT_PARSER = createParser(false);
+
+    @SuppressWarnings("unchecked")
+    private static ConstructingObjectParser<Hyperparameters, Void> createParser(boolean ignoreUnknownFields) {
+        ConstructingObjectParser<Hyperparameters, Void> parser = new ConstructingObjectParser<>(NAME,
+            ignoreUnknownFields,
+            a -> new Hyperparameters((String)a[0], (Double)a[1], (Double)a[2], (Double)a[3], (Boolean)a[4]));
+        parser.declareString(ConstructingObjectParser.constructorArg(), HYPERPARAMETER_NAME);
+        parser.declareDouble(ConstructingObjectParser.constructorArg(), VALUE);
+        parser.declareDouble(ConstructingObjectParser.constructorArg(), ABSOLUTE_IMPORTANCE);
+        parser.declareDouble(ConstructingObjectParser.constructorArg(), RELATIVE_IMPORTANCE);
+        parser.declareBoolean(ConstructingObjectParser.constructorArg(), SUPPLIED);
+        return parser;
+    }
+
+    public static Hyperparameters fromXContent(XContentParser parser, boolean lenient) throws IOException {
+        return lenient ? LENIENT_PARSER.parse(parser, null) : STRICT_PARSER.parse(parser, null);
+    }
+
+    public final String hyperparameterName;
+    public final Double value;
+    public final Double absoluteImportance;
+    public final Double relativeImportance;
+    public final Boolean supplied;
+
+    public Hyperparameters(StreamInput in) throws IOException {
+        this.hyperparameterName = in.readString();
+        this.value = in.readDouble();
+        this.absoluteImportance = in.readDouble();
+        this.relativeImportance = in.readDouble();
+        this.supplied = in.readBoolean();
+    }
+
+    Hyperparameters(String hyperparameterName, Double value, Double absoluteImportance, Double relativeImportance, Boolean supplied) {
+        this.hyperparameterName = hyperparameterName;
+        this.value = value;
+        this.absoluteImportance = absoluteImportance;
+        this.relativeImportance = relativeImportance;
+        this.supplied = supplied;
```

Comment:
If the `Double`s and the `Boolean`  are truly not nullable, they should be `double` and `boolean` (unboxed, not nullable). 

If they are nullable, the parser needs to treat them as optional and the serialization needs to as well